app/controllers/DraftMetadataChecksResultsController.scala

```diff
@@ -27,12 +31,31 @@ class DraftMetadataChecksResultsController @Inject() (
     if (applicationConfig.blockDraftMetadataUpload) {
       Future(Ok(views.html.notFoundError(name = request.token.name, isLoggedIn = true, isJudgmentUser = false)))
     } else {
+      val token = request.token.bearerAccessToken
       for {
         reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+        consignmentStatuses <- consignmentStatusService.getConsignmentStatuses(consignmentId, token)
       } yield {
-        Ok(views.html.draftmetadata.draftMetadataChecksResults(consignmentId, reference, frontEndInfoConfiguration.frontEndInfo, request.token.name))
+        Ok(
+          views.html.draftmetadata
+            .draftMetadataChecksResults(consignmentId, reference, getValue(consignmentStatuses, DraftMetadataType), frontEndInfoConfiguration.frontEndInfo, request.token.name)
+        )
           .uncache()
       }
     }
   }
+
+  def getValue(statuses: List[GetConsignment.ConsignmentStatuses], statusType: StatusType): DraftMetadataProgress = {
+    val failed = DraftMetadataProgress("FAILED", "red")
+    statuses.find(_.statusType == statusType.id).map(_.value).map {
+      case FailedValue.value              => failed
+      case CompletedWithIssuesValue.value => DraftMetadataProgress("ERRORS", "red")
+      case CompletedValue.value           => DraftMetadataProgress("IMPORTED", "blue")
+      case _                              => failed
+    } getOrElse failed
+  }
+}
+
+object DraftMetadataChecksResultsController {
+  case class DraftMetadataProgress(value: String, colour: String)
```

Comment:
You can simply create this `DraftMetadataProgress` case class without the `object DraftMetadataChecksResultsController`.